Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -651,25 +689,39 @@ public virtual void OnPointerUp(MixedRealityPointerEventData eventData)
             Vector3 angularVelocity = GetPointersAngularVelocity();
 
             uint id = eventData.Pointer.PointerId;
-            if (pointerIdToPointerMap.ContainsKey(id))
+            PointerData pointerDataToRemove = default;
+            bool removePointerData = false;
+            int pointerDataListCount = pointerDataList.Count;
+            for (int i = 0; i < pointerDataListCount; i++)
```

Comment:
Maybe all of these for loops can be consolidated to a helper function rather than many for loops.

Reply:
Consolidated! These three, at least. The rest didn't follow a reasonably similar enough pattern.